src/PacketaWidget/useTranslateCountry.js

```diff
@@ -37,6 +38,17 @@ export const useTranslateCountry = (
 			} else {
 				const previousCountryName = dynamicSettings.countryName;
 				if ( previousCountryName !== countryName ) {
+					const updateSettings = function ( key, value ) {
```

Comment:
Myslím, že tohle je zbytečně komplikovaná konstrukce. Tohle podle mě bude fungovat úplně stejně:
```
				if ( previousCountryName !== countryName ) {
					setDynamicSettings( { ...dynamicSettings, countryName } );
					setViewState( null );

					if ( ! loading ) {
						setLoading( true );
						fetch( adminAjaxUrl, {
							method: 'POST',
							headers: {
								'Content-Type': 'application/x-www-form-urlencoded',
							},
							body: new URLSearchParams( {
								action: 'translate_country_name',
								countryName: countryName,
							} ),
						} )
							.then( ( response ) => response.json() )
							.then( ( country ) => {
								if ( country !== null ) {
									setDynamicSettings(prevState => ( {...prevState, country } ) );
								}
							} )
							.catch( ( error ) => {
								console.error( 'Error:', error );
								// keep previous country
							} )
							.finally( () => {
								setLoading( false );
							} );
					}
				}
```

To první volání nad fetchem nepotřebuje složitě přistupovat k prevState, protože aktuální state má k dispozici v `dynamicSettings` (nejedná se o asynchronní kód). Vynullování `viewState` se může udělat jen jednou tady, protože nic než `null` se tam stejně nikde později nenastavuje.
Co mě trochu znervózňuje je stav, kdy se do `dynamicSettings` na začátku nastaví `countryName`, ale pokud fetch selže, tak v `country` bude podle všeho neplatný obsah a tyhle dvě nastavení budou v rozporu. Ale nevím, jaké je validní řešení takového stavu.

Reply:
Dobré návrhy, zapracováno. Znervóznění chápu, nicméně je to daň za to, že na to WC nemá endpoint. Prostě zatím nikdo nemyslel na pokročilejší shipping pluginy.